kv/kvserver: TestReplicateQueueRebalanceMultiStore failed

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/3ae4c75f-03bf-474e-9fe7-d7c024fbe0dc?testReportRun=26&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [45cd83b6838bd515c273955c90ade06ca9fa8794](https://github.com/cockroachdb/cockroach/commits/45cd83b6838bd515c273955c90ade06ca9fa8794):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestReplicateQueueRebalanceMultiStore (14m41s)
	TestReplicateQueueRebalanceMultiStore/multi-store (13m56s)

```
Stack: 

```
goroutine 27391982 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 6 9 8 7]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [35 34 31 28 34 30 34 29]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [18 17 8 9 7 9 8 7]
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=26</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35687

kv/kvserver.TestReplicateQueueRebalanceMultiStore [failed](https://tanzanite.cluster.engflow.com/invocations/default/9bf21f38-47fd-44ee-9ab1-67dabe38b27d?testReportRun=9&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cva3Yva3ZzZXJ2ZXI6a3ZzZXJ2ZXJfdGVzdA==) on master @ [24547fb1115903d432004207fc49a87262133c34](https://github.com/cockroachdb/cockroach/commits/24547fb1115903d432004207fc49a87262133c34):

Fatal error:

```
panic: test timed out after 14m57s
running tests:
	TestReplicateQueueRebalanceMultiStore (14m38s)
	TestReplicateQueueRebalanceMultiStore/multi-store (13m40s)

```
Stack: 

```
goroutine 26429584 [running]:
testing.(*M).startAlarm.func1()
	GOROOT/src/testing/testing.go:2259 +0x3b9
created by time.goFunc
	GOROOT/src/time/sleep.go:176 +0x2d
```
<details><summary>Log preceding fatal error</summary>
<p>

```
=== RUN   TestReplicateQueueRebalanceMultiStore
    test_log_scope.go:170: test logs captured to: outputs.zip/logTestReplicateQueueRebalanceMultiStore1917980383
    test_log_scope.go:81: use -show-logs to present logs inline
=== RUN   TestReplicateQueueRebalanceMultiStore/multi-store
    replicate_queue_test.go:278: split off r68:/{Table/100-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=9, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r69:/{Table/101-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=10, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r70:/{Table/102-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=11, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r71:/{Table/103-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=12, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r72:/{Table/104-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=13, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r73:/{Table/105-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=14, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r74:/{Table/106-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=15, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r75:/{Table/107-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=16, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r76:/{Table/108-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=17, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r77:/{Table/109-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=18, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r78:/{Table/110-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=19, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r79:/{Table/111-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=20, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r80:/{Table/112-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=21, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r81:/{Table/113-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=22, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r82:/{Table/114-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=23, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:278: split off r83:/{Table/115-Max} [(n1,s2):1, (n2,s3):2, (n4,s7):3, next=4, gen=24, sticky=9223372036.854775807,2147483647]
    replicate_queue_test.go:334: current replica state (want at least 28 replicas on all stores): [31 31 35 31 29 30 34 33]
    replicate_queue_test.go:345: current lease state (want at least 7 leases on all stores): [15 8 14 8 9 9 9 10]
```
</p>
</details>

Parameters:
 - <code>attempt=1</code>
 - <code>run=9</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestReplicateQueueRebalanceMultiStore.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -223,8 +223,10 @@ func TestReplicateQueueRebalanceMultiStore(t *testing.T) {
 		t.Run(testCase.name, func(t *testing.T) {
 			if testCase.storesPerNode > 1 {
 				// 8 stores with active rebalancing can lead to failed heartbeats due
-				// to overload. Skip under stress when running the multi-store variant.
+				// to overload. Skip under stress and remote  execution when running
+				// the multi-store variant.
 				skip.UnderStress(t)
+				skip.UnderRemoteExecutionWithIssue(t, 118347)
 			}
 			// Set up a test cluster with multiple stores per node if needed.
 			args := base.TestClusterArgs{

